bpf/maps.go

```diff
@@ -222,6 +222,15 @@ func (b *PinnedMap) Delete(k []byte) error {
 	cmd := exec.Command("bpftool", args...)
 	out, err := cmd.CombinedOutput()
 	if err != nil {
+		if strings.Contains(string(out), "No such file or directory") {
+			// Check that the error wasn't about the map as a whole being missing.
+			if _, statErr := os.Stat(b.versionedFilename()); statErr == nil {
```

Comment:
The error already tells you what sort of ENOENT it is, doesnt it?

```
[root@suez /]# bpftool -j -p map delete pinned /sys/fs/bpf/tc/globals/cali_v4_nat_aff key 01 02 03 04 05 06 07 01 02 03 04 05 06 07 01 01
{
    "error": "delete failed: No such file or directory"
}
[root@suez /]# bpftool -j -p map delete pinned /sys/fs/bpf/tc/globals/cali_v4_nat_aff123 key 01 02 03 04 05 06 07 01 02 03 04 05 06 07 01 01
{
    "error": "bpf obj get (/sys/fs/bpf/tc/globals/cali_v4_nat_aff123): No such file or directory"
}
```

Reply:
Didn't spot that, will respin